### ISSUE TITLE
Unexpected valid docs warning

### ISSUE THREAD

**Issue by mAu888** [realm/SwiftLint#323]
The following documentation unexpectedly triggers a `valid_docs` warning. Am I missing something? Given the `nonTriggeringExamples` in the `ValidDocsRule.swift` I expect it to not trigger in this case.

``` swift
    /**
     Returns the first element matching `includeElement`
     - parameter includeElement: Predicate to match
     - throws: Rethrows the error of `includeElement`
     - returns: First element matching the predicate or `nil`
    */
    func first(@noescape includeElement: (Self.Generator.Element) throws -> Bool) rethrows -> Self.Generator.Element? {
        for el in self where try includeElement(el) {
            return el
        }

        return nil
    }
```

**Comment by norio-nomura** [comment]
It seems sourcekitd.framework fails to detecting function's parameter when using both of `@noescape` and `rethrows`. https://github.com/jpsim/SourceKitten/issues/133
That makes SwiftLint to detect `- parameter includeElement:` as a surplus and generate `valid_docs` warning.

**Comment by mAu888** [comment]
Interesting. Thanks for forwarding the issue :+1:

**Comment by jpsim** [comment]
Thanks for filing, and thanks to @norio-nomura for re-filing on bugs.swift.org

**Comment by marcelofabri** [comment]
Closing this since we removed this rule.
